packages/qtile-git/qtile-git.pacscript

```diff
@@ -0,0 +1,39 @@
+name="qtile-git"
+pkgname="qtile"
+url="https://github.com/qtile/qtile.git"
+license="MIT"
+depends="python3-cffi xserver-xorg python3-xcffib python3-cairocffi libpangocairo-1.0-0 python3-pip python3-dbus libpulse-dev pavucontrol"
+description="A full-featured, hackable tiling window manager written and configured in Python"
+maintainer="Ruturajn <nanotiruturaj@gmail.com>"
+
+pkgver() {
+  git ls-remote "${url}" master | cut -f1
+}
+version="$(pkgver)"
+
+prepare() {
+  true
+}
+
+build() {
+  true
+}
+
+install() {
+  pip install --upgrade --no-cache --force-reinstall cairocffi
+  pip3 install dbus-next
+  pip3 install .
+  sudo python3 setup.py install
+  sudo mkdir -p /usr/src/pacstall/qtile-git/usr/share/xsessions
+  echo "[Desktop Entry]
+  Name=Qtile
+  Comment=Qtile Session
+  Exec=qtile start
+  Type=Application
+  Keywords=wm;tiling" | sudo tee -a /usr/src/pacstall/qtile-git/usr/share/xsessions/qtile.desktop > /dev/null
+}
+
+
+removescript() {
+  sudo rm -f /usr/local/bin/qtile
+}
```

Comment:
@Henryws is this really needed?

Reply:
I added it since it remains after I remove the package, and you suggested that I add it.

Reply:
![image](https://user-images.githubusercontent.com/56625259/159960633-b14ea65a-a081-49d8-a35c-de86827613f6.png)

Reply:
This is the output from ```pacstall -T qtile-git```

Reply:
I think that the file ```qtile.desktop``` is being created in ```/usr/share/xsessions/``` automatically, without explicitly doing it in the pacscript.

Reply:
What do I need to change then?

Reply:
Yes, because of a previous build

Reply:
So, should I remove the ```removescript()``` function?

Reply:
Figure out how to build it to the proper location first.

Reply:
Where should it be ideally?

Reply:
Ok thanks, I will try it out this way, and about the directory should it be the ```$STOWDIR```?

Reply:
`${STOWDIR}/${pkgname}`

Reply:
Ok

Reply:
I made the following changes,
```
install() {
  pip install --upgrade --no-cache --force-reinstall cairocffi
  pip3 install dbus-next
  sudo mkdir -p /usr/src/pacstall/qtile-git/usr/share/xsessions
  sudo pip3 install --target=/usr/src/pacstall/qtile-git .
  sudo python3 setup.py build -b /usr/src/pacstall/qtile-git install
  echo "[Desktop Entry]
  Name=Qtile
  Comment=Qtile Session
  Exec=qtile start
  Type=Application
  Keywords=wm;tiling" | sudo tee -a /usr/src/pacstall/qtile-git/usr/share/xsessions/qtile.desktop > /dev/null
}
```
But, I am getting this error:
```
Finished processing dependencies for qtile==0.21.0
[+] INFO: Symlinking files
WARNING! stowing qtile-git would cause conflicts:
  * existing target is not owned by stow: bin
All operations aborted.
[!] ERROR: Package contains links to files that exist on the system
[+] INFO: Cleaning up
[!] ERROR: Failed to install qtile-git

```

Reply:
Ok

Reply:
[qtile-test.txt](https://github.com/pacstall/pacstall-programs/files/8347217/qtile-test.txt)

I have directed the output of ```tree /tmp/qtile-git/``` to this file.

Reply:
Even after passing the target directories to setup.py and pip, ```/usr/local/bin/qtile```, is being created.

Reply:
I changed the order for ```setup.py``` and ```pip```,
```
install() {
  pip install --upgrade --no-cache --force-reinstall cairocffi
  pip3 install dbus-next
  sudo mkdir -p /usr/src/pacstall/qtile-git/usr/share/xsessions
  sudo python3 setup.py build -b /usr/src/pacstall/qtile-git install
  sudo pip3 install --target=/usr/src/pacstall/qtile-git/ . 
  echo "[Desktop Entry]
  Name=Qtile
  Comment=Qtile Session
  Exec=qtile start
  Type=Application
  Keywords=wm;tiling" | sudo tee -a /usr/src/pacstall/qtile-git/usr/share/xsessions/qtile.desktop > /dev/null
}
```
Now, there is no ```/usr/local/bin/qtile``` file created, but the ```stow``` error remains.

Reply:
I am doing it like this now,
```
install() {
  pip install --upgrade --no-cache --force-reinstall cairocffi
  pip3 install dbus-next
  sudo mkdir -p /usr/src/pacstall/qtile-git/usr/share/xsessions
  sudo python3 setup.py build -b /usr/src/pacstall/qtile-git/usr install
  sudo pip3 install . --target=/usr/src/pacstall/qtile-git/usr/ 
  echo "[Desktop Entry]
  Name=Qtile
  Comment=Qtile Session
  Exec=qtile start
  Type=Application
  Keywords=wm;tiling" | sudo tee -a /usr/src/pacstall/qtile-git/usr/share/xsessions/qtile.desktop > /dev/null
}
```
And now I get this,
![Screenshot from 2022-03-25 11-47-43](https://user-images.githubusercontent.com/56625259/160065454-5dc2ee1f-2853-4fa3-bd94-4ca59fff9474.png)

Reply:
I changed the ```install()``` function to this,
```
install() {
  pip install --upgrade --no-cache --force-reinstall cairocffi
  pip3 install dbus-next
  sudo mkdir -p /usr/src/pacstall/qtile-git/usr/share/xsessions
  sudo pip3 install . --target=/usr/src/pacstall/qtile-git/usr
  echo "[Desktop Entry]
  Name=Qtile
  Comment=Qtile Session
  Exec=qtile start
  Type=Application
  Keywords=wm;tiling" | sudo tee -a /usr/src/pacstall/qtile-git/usr/share/xsessions/qtile.desktop > /dev/null
}
```

I checked the Qtile Documentation, where it's mentioned that only ```pip install``` needs to be done after cloning the git repository : http://docs.qtile.org/en/stable/manual/install/index.html#installing-from-source

And, now only ```/usr/bin/qtile``` is created
![Screenshot from 2022-03-25 14-07-37](https://user-images.githubusercontent.com/56625259/160085619-cc94b24a-ea5d-404c-bd52-2f4d2e3111ef.png)

This is what was expected right?

Reply:
I have changed the ```install()``` function to this,
```
install() {
  sudo pip3 install --upgrade --no-cache --force-reinstall cairocffi cffi xcffib
  sudo pip3 install dbus-next
  sudo mkdir -p /usr/src/pacstall/qtile-git/usr/share/xsessions
  sudo python3 setup.py build -b /usr/src/pacstall/qtile-git/usr install --root=/usr/src/pacstall/qtile-git/
  sudo cp ./resources/qtile.desktop /usr/src/pacstall/qtile-git/usr/share/xsessions/qtile.desktop
}
```
The ```qtile.desktop``` file is already present in qtile's git repo, so I copied it from there, and ```setup.py install``` and ```pip install .``` do the basically the same thing, and since using ```setup.py install``` allows for easier control over the target installation directory, I have used that.

![2022-03-25_21-38](https://user-images.githubusercontent.com/56625259/160159419-67303de9-0480-43d5-91be-038703ca59ef.png)

![2022-03-25_21-39](https://user-images.githubusercontent.com/56625259/160159470-5f23c465-d82f-427f-8456-0e8441d42624.png)

This is after ```pacstall -R qtile-git```,

![2022-03-25_21-40](https://user-images.githubusercontent.com/56625259/160159505-28e15691-a6e9-492b-b971-193b86a6dba8.png)

So, should I commit this?

Reply:
Run `pacstall -R qtile-git` with your fix

Reply:
Yeah, I have attached the screenshot for it. Should I attach the whole output?

Reply:
Send a file

Reply:
Oh, yeah btw I removed the ```removescript()``` function since, ```pacstall -R``` was handling everything

Reply:
[qtile-test-out.txt](https://github.com/pacstall/pacstall-programs/files/8352832/qtile-test-out.txt)

Reply:
I meant the output of `pacstall -R qtile-git`

Reply:
Yeah I have directed the output from that command to this file

Reply:
Oh, sorry!! I meant `-T`

Reply:
ohh ok

Reply:
[qtile-pacstall-T-qtile-git.txt](https://github.com/pacstall/pacstall-programs/files/8352876/qtile-pacstall-T-qtile-git.txt)

Reply:
Running your pacscript returns this with `pacstall -T qtile-git`
```bash
/
`-- usr
    `-- share
        `-- xsessions
            `-- qtile.desktop

3 directories, 1 file
```

Reply:
I think you have used the original pacscript which did not have the fix that I have provided above. Could you try it out with this : [qtile-git-new-pacscript.txt](https://github.com/pacstall/pacstall-programs/files/8354605/qtile-git-new-pacscript.txt)

Reply:
Ok, it worked in the docker, testing on my main machine. 🤞🏻

Reply:
Yeah, I am thinking about how to fix that

Reply:
Is this fine ?
[pacstall-T-qtile-git-new.txt](https://github.com/pacstall/pacstall-programs/files/8355795/pacstall-T-qtile-git-new.txt)

Reply:
Send me the new pacscript. I can't test your fix without it.

Reply:
I removed the build command in the ```setup.py install```

Reply:
![image](https://user-images.githubusercontent.com/58742515/160247156-c224d1b0-5d49-47b7-ac0c-be15d2f723c3.png)
this line?

Reply:
Yeah

Reply:
Cool that works